### DISCUSSION TITLE
Possible Bruno Backstage Plugin

### DISCUSSION THREAD

**Discussion by JSSAggie** [usebruno/bruno#2884] [Ideas]
Our team were heavy Postman users from the very beginning when postman first offered a paid offering.

As of March this year we no longer offer a paid license to developers.  We backed up all the postman collections, into git, and teams have begun to take those backups and convert them to bruno collections.  The bruno collections are co-located within the same code repository as their service code.

When a service API changes the developer also changes their bruno collection when needed.  One other detail that is worth adding is we are an "API Design first" shop and propose API service changes by raising a pull request to our specification repositories and then once approved developers move forward with making the service code changes.

When other developers around the organization would like to consume an API they discover they then pull down the service code along with the bruno collection to make their first call.

We are also heavy users of backstage to discover services and APIs that teams might want to consume.  We surface in backstage every service, the API definition (open api spec), and links to the service repository.

We are thinking about ways we could reduce the "time to first call" of an API you might discover in backstage.

We wanted to first see if others are interested in the same types of workflows including backstage?

If there is interest we would be open to collaborating on something?

**Brainstorming / Thoughts:**

- It is possible for us to update our CI/CD process to publish bruno collections as some sort of artifact that could be linked to the service in backstage.
- How could we go from a link in backstage to downloading the artifact and launching bruno with the collection loaded.
- How would we handle updates when that collection changes by the service team?

**Comment by JSSAggie** [comment]
This might be something comparable:
https://github.com/coder/backstage-plugins/blob/main/plugins/backstage-plugin-devcontainers-react/README.md

**Comment by berlingoqc** [comment]
hey I'm interested in this. Looking to do something similar in my organization.

**Comment by berlingoqc** [comment]
adding support for bruno to load a tarball of a collection from a launch via the browser could be usefull for this.

**Reply by berlingoqc** [reply]
https://www.electronjs.org/docs/latest/api/app#event-open-url-macos

and create an handler to download tarball , extract to wanted location and you are ready to go.

After for the update it's more tricky, ideally, some versioning would be needed inside the collection to compare and refresh against a server tarball. A bit like helm chart repository are working

**Comment by berlingoqc** [comment]
I'll do this issue , to create a exporter via bru.js cli tool
https://github.com/usebruno/bruno/issues/2679
with this you can export a single json file for your collection.

and with this we could load the url into bruno
https://github.com/usebruno/bruno/issues/2970

we just need a third issue to add support for open-url afterward
bruno:://collections-import?url=.....

**Reply by JSSAggie** [reply]
Looks like you created #3000 to cover the last bit.  Nice work.

The way I see this working with what you have going on is that backstage would have the full collection JSON and provide a link to launch bruno with a reference to that full json collection.

This looks solid for a v1.

**Comment by shaky2k15** [comment]
Any thoughts when this bruno integration plugin is expected in backstage release ?

**Comment by gaelgoth** [comment]
I’m also interested in a Backstage plugin. Maybe it’s something to request at https://github.com/backstage/community-plugins?

**Comment by abhishek-bruno** [comment]
Hi all,
A quick update from our side. We're building an official, open-source Bruno plugin for Backstage, and there's an RFC open at #9208 on how collections should be modeled in the catalog. Have a look and let us know which one fits your setup.

Want to talk it through? Feel free to [book a call](https://cal.com/abhishek-bruno/30min) with me.
